src/tiled/tileseteditor.cpp

```diff
@@ -465,7 +465,28 @@ void TilesetEditor::addTiles()
         Tile *newTile = new Tile(tileset->takeNextTileId(), tileset);
         newTile->setImage(loadedFile.image);
         newTile->setImageSource(loadedFile.imageSource);
-        tiles.append(newTile);
+
+        // If the tile is already in the tileset, warn user and confirm addition
+        bool addImage = true;
+        const QMap<int, Tile*> addedTiles = mCurrentTilesetDocument->tileset().data()->tiles();
+        for(auto tile : addedTiles) {
+            if(tile->imageSource() == newTile->imageSource()) {
+                QMessageBox warning(QMessageBox::Warning,
+                            tr("Add Tiles"),
+                            tr("Tile already exists in the Tileset!"),
```

Comment:
Which tile exists already? Since the user may have selected multiple files, you need to include the file name as done in the "Could not load" message box above.